misc/appmesh-plugin-validator/appmesh-plugin-validator.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/coreos/go-iptables/iptables"
+)
+
+const (
+	ingressChain               = "APPMESH_INGRESS"
+	egressChain                = "APPMESH_EGRESS"
+	gid                        = "133"
+	appPorts                   = "9080,9090"
+	proxyEgressPort            = "15001"
+	proxyIngressPort           = "15000"
+	egressIgnoredPorts         = "80,81"
+	egressIgnoredIPv4Addresses = "192.168.100.0/22,163.107.163.107,169.254.170.2,169.254.169.254"
+)
+
+func main() {
+	if err := validateIPRules(); err != nil {
+		fmt.Errorf("Unable to invoke aws-appmesh cni plugin: %v\n", err)
+		os.Exit(1)
+	}
+	os.Exit(42)
+}
+
+// validateIPRules validates IP rules created in the container network namespace.
+func validateIPRules() error {
+	var iptable *iptables.IPTables
+	var err error
+
+	if iptable, err = iptables.NewWithProtocol(iptables.ProtocolIPv4); err != nil {
+		return err
+	}
+
+	// Print detailed info for debugging.
```

Comment:
were you planning to use awslogs to surface the output? i didn't see you use it in functional test

Reply:
I haven't thought about this. So storing log on the awslog is for easier to debug?

Reply:
substantially.